Marlin/src/module/planner.h

```diff
@@ -559,9 +559,37 @@ class Planner {
       #if ENABLED(FWRETRACT)
         apply_retract(raw);
       #endif
+
       #if IS_KINEMATIC
+        #if HAS_FEEDRATE_SCALING
+          // DELTA/SCARA need to recalculate the feedrate in
+          // order to complete the move in the given time.
+          const float inv_segment_length = 1.0f / millimeters,
+                      inverse_secs = inv_segment_length * fr_mm_s,
+                      oldA = position_float[A_AXIS],
+                      oldB = position_float[B_AXIS]
+                      #if ENABLED(DELTA_FEEDRATE_SCALING)
+                        , oldC = position_float[C_AXIS]
+                      #endif
```

Comment:
The `oldA`, `oldB`, `oldC` values are not cartesian XYZ. They are either the delta tower positions (mm) or the SCARA elbow angles (degrees).